protos/datamodel/costs.proto

```diff
@@ -0,0 +1,236 @@
+  // Load balancer identifier
+  string loadBalancerId = 1;
+
+  // Associated service name
+  string serviceName = 2;
+
+  // Load balancer IP address
+  string ipAddress = 3;
+
+  // Whether this is a private load balancer
+  bool isPrivate = 4;
+
+  // Load balancer cost
+  CostMetrics cost = 5;
+
+  // Hours the load balancer was active
+  float hours = 6;
+}
+
+// AllocationCost represents complete cost breakdown for a resource allocation
+// This is the main cost container that gets embedded in resource messages
+message AllocationCost {
+  // Window for this cost calculation (inherits from cluster window if not specified)
+  TimeWindow window = 1;
+
+  // Resource-specific costs
+  ResourceCost cpu = 2;
+  ResourceCost ram = 3;
+
+  // GPU costs (can have multiple GPUs)
+  repeated GPUCostDetail gpu = 4;
+
+  // Storage costs (persistent volumes)
+  repeated StorageCostDetail storage = 5;
+
+  // Network costs with detailed breakdown
+  DetailedNetworkCost network = 6;
+
+  // Load balancer costs
+  repeated LoadBalancerCostDetail loadBalancers = 7;
+
+  // Shared costs allocated to this resource
+  CostMetrics shared = 8;
+
+  // External cloud costs allocated to this resource
+  CostMetrics external = 9;
+
+  // Pre-computed total cost for performance (sum of all above)
+  CostMetrics total = 10;
+}
+
+// CostAttribution represents hierarchical cost attribution information
+// Maps to OpenCost's AllocationProperties for cost attribution
+message CostAttribution {
+  // Cluster-level attribution
+  string cluster = 1;
+  string providerId = 2;
+
+  // Node-level attribution
+  string node = 3;
+
+  // Namespace-level attribution
+  string namespace = 4;
+  map<string, string> namespaceLabels = 5;
+  map<string, string> namespaceAnnotations = 6;
+
+  // Workload-level attribution
+  string controller = 7;
+  string controllerKind = 8; // deployment, statefulset, daemonset, job
+
+  // Pod-level attribution
+  string pod = 9;
+
+  // Container-level attribution
+  string container = 10;
+
+  // Service-level attribution
+  repeated string services = 11;
+
+  // Custom attribution
+  map<string, string> labels = 12;
+  map<string, string> annotations = 13;
+}
+
+// SharedCostAllocation represents shared/overhead costs
+// For costs that need to be distributed across multiple resources
+message SharedCostAllocation {
+  // Identifier for the shared cost pool
+  string costPoolId = 1;
+
+  // Name/description of the shared cost
+  string name = 2;
+
+  // Total shared cost metrics
+  CostMetrics totalCost = 3;
+
+  // Sharing algorithm used (weighted, even, proportional)
+  string sharingMethod = 4;
+
+  // Window for this shared cost
+  TimeWindow window = 5;
+
+  // Cost attribution for this shared cost
+  CostAttribution attribution = 6;
+}
+
+// ExternalCostAllocation represents external cloud service costs
+// For costs from cloud billing APIs not directly tied to K8s resources
+message ExternalCostAllocation {
```

Comment:
We can delete this. It's a deprecated feature. (If I'm understanding the name correctly.) At any rate, the agent would not compute these -- "external costs" was a server-side concept applied at query-time.